### PR TITLE
Harmonize uid check regexp - accept dot and dash in the frontend (#2448)

### PR DESCRIPTION
**Pull request by landryb**
while here properly escape dot and dash in the serverside validation.

i'm not 100% sure about escaping - at the end of a class.. but pretty sure dot needs to be escaped, otherwise the regexp reduces to `[a-z].*` i think...

### CODE REVIEW

**Comment by landryb** [comment]
And remember i'm no java expert either - so might be wrong on the dot escaping, but anyway the two regexps don't match - and that's the bug i want to squash here, accepting dot and dash :)

File: console/src/main/webapp/manager/app/templates/userForm.tpl.html [hunk 1]
```diff
@@ -108,8 +108,8 @@
           {{model.uid}} (<a href="javascript:void(0)" ng-click="view.edit_login=!view.edit_login">change</a>)
         </div>
         <div class="col-sm-8" ng-if="view.edit_login">
-          <!-- Server validation should match this pattern: /^[a-z]+\w*$/ -->
-          <input ng-model="model.uid" class="form-control" id="uid" placeholder="{{'user.login' | translate}}" ng-required="required.uid" ng-pattern="'^[a-z]+\\\\w*$'">
+          <!-- Server validation is in java/org/georchestra/console/ws/newaccount/NewAccountFormController.java -->
+          <input ng-model="model.uid" class="form-control" id="uid" placeholder="{{'user.login' | translate}}" ng-required="required.uid" ng-pattern="'^[a-zA-Z][a-zA-Z0-9\\\\.\\\\-]*$'">
```
Comment (tonio) [inline console/src/main/webapp/manager/app/templates/userForm.tpl.html]:
```suggestion
          <input ng-model="model.uid" class="form-control" id="uid" placeholder="{{'user.login' | translate}}" ng-required="required.uid" ng-pattern="/^[a-z][a-z0-9\\\\.\\\\-]*$/i">
```
(untested but [should work](https://docs.angularjs.org/api/ng/directive/ngPattern))

**Comment by landryb** [comment]
After a bit of testing on https://www.freeformatter.com/java-regex-tester.html it seems dot and dash don't need escaping in java. weird. my interpretation of http://tutorials.jenkov.com/java-regex/index.html#escaping-characters lead me to believe the opposite..

\w in js includes underscore, so underscore should be added to the serverside regexp anyway if it's accepted in the frontend.

**Comment by fvanderbiest** [comment]
Thanks Landry

**Pull request merged**